src/sentry/api/endpoints/organization_integration_details.py

```diff
@@ -4,27 +4,42 @@
     OrganizationEndpoint, OrganizationIntegrationsPermission
 )
 from sentry.api.serializers import serialize
-from sentry.models import Integration, OrganizationIntegration
+from sentry.models import OrganizationIntegration, ProjectIntegration
 
 
 class OrganizationIntegrationDetailsEndpoint(OrganizationEndpoint):
     permission_classes = (OrganizationIntegrationsPermission, )
 
     def get(self, request, organization, integration_id):
-        integration = Integration.objects.get(
-            organizations=organization,
-            id=integration_id,
+        integration = OrganizationIntegration.objects.get(
+            integration_id=integration_id,
+            organization=organization,
         )
 
         return self.respond(serialize(integration, request.user))
 
     def delete(self, request, organization, integration_id):
-        integration = Integration.objects.get(
-            organizations=organization,
-            id=integration_id,
-        )
+        # Removing the integration removes both the organization and project
+        # integration.
         OrganizationIntegration.objects.filter(
-            integration=integration,
+            integration_id=integration_id,
             organization=organization,
         ).delete()
+        ProjectIntegration.objects.filter(
+            integration_id=integration_id,
+            project__organization=organization,
+        ).delete()
+
         return self.respond(status=204)
+
+    def post(self, request, organization, integration_id):
+        integration = OrganizationIntegration.objects.get(
+            integration_id=integration_id,
+            organization=organization,
+        )
+
```

Comment:
you can easily put integration to a `def get_object()` or something like that as organization is in the kwargs, and you can get integraiton_id too frrom there rather than repeating the code here and there and for the next n ones.